### PR TITLE
Remove FormNumberType

### PR DESCRIPTION
**Pull request by vikmeup**

### CODE REVIEW

File: Formalist/FormNumberFormatter.swift [hunk 1]
```diff
@@ -6,28 +6,22 @@
 //  Copyright © 2017 Seed Platform, Inc. All rights reserved.
 //
 
-public enum FormNumberType {
-    case EIN
-    case SSN
-    case creditCard
```
Comment (klaaspieter) [inline Formalist/FormNumberFormatter.swift]: I think credit card is common enough to have a special field for. I like the [`DateEncodingStrategy`](https://github.com/apple/swift/blob/master/stdlib/public/SDK/Foundation/JSONEncoder.swift#L40) Apple uses. The custom patterns could be expressed using `case custom((String) -> String) -> Void)`.

File: Example/ViewController.swift [hunk 2]
```diff
@@ -35,7 +35,8 @@ class ViewController: UIViewController {
         let edgeInsets = UIEdgeInsets(top: 15, left: 15, bottom: 15, right: 15)
 
         let numberFormatter = FormNumberFormatter(
-            type: .custom(pattern: "***-***-****", replaceCharacter: "*")
+            pattern: "***-***-****",
```
Comment (klaaspieter) [inline Example/ViewController.swift]: Could this become a problem for user's who'd want to use literal `*` in their number format?

File: Formalist/FormNumberFormatter.swift (hunk 1 above)
Comment (grooot-ios) [inline Formalist/FormNumberFormatter.swift]: Chatted with @louoso, sounds like there is different type of credit cards, let's use generic pattern that use can set for now. 

pasted image at 2017_07_25 03_38 pm

File: Example/ViewController.swift (hunk 2 above)
Comment (grooot-ios) [inline Example/ViewController.swift]: FormNumberFormatter only allows to enter numbers.

File: Formalist/FormNumberFormatter.swift (hunk 1 above)
Comment (vikmeup) [inline Formalist/FormNumberFormatter.swift]: Chatted with @louoso, sounds like there is different type of credit cards, let's use generic pattern that use can set for now. 

pasted image at 2017_07_25 03_38 pm

File: Example/ViewController.swift (hunk 2 above)
Comment (vikmeup) [inline Example/ViewController.swift]: FormNumberFormatter only allows to enter numbers.

**Pull request merged**